pynapple/core/ts_group.py

```diff
@@ -1030,6 +1030,160 @@ def getby_category(self, key):
+
+        >>> dict2 = {10: nap.Ts(t=[-1, 0, 1], time_units='s')}
+        >>> tsgroup2 = nap.TsGroup(dict2, time_support=time_support_a)
+
+        >>> dict3 = {0: nap.Ts(t=[-.1, 0, .1], time_units='s')}
+        >>> tsgroup3 = nap.TsGroup(dict3, time_support=time_support_a)
+
+        >>> dict4 = {10: nap.Ts(t=[-1, 0, 1], time_units='s')}
+        >>> tsgroup4 = nap.TsGroup(dict2, time_support=time_support_b)
+
+        Merge with default options if have same time_support and non-overlapping indexes:
+
+        >>> tsgroup_12 = nap.TsGroup.merge_group(tsgroup1, tsgroup2)
+        >>> tsgroup_12
+        Index    rate
+        -------  ------
+              0     1.5
+             10     1.5
+
+        Pass reset_index=True if indexes are overlapping:
+
+        >>> tsgroup_13 = nap.TsGroup.merge_group(tsgroup1, tsgroup3, reset_index=True)
+        >>> tsgroup_13
+
+        Index    rate
+        -------  ------
+              0     1.5
+              1     1.5
+
+        Pass reset_time_support=True if time_supports are different:
+
+        >>> tsgroup_14 = nap.TsGroup.merge_group(tsgroup1, tsgroup4, reset_time_support=True)
+        >>> tsgroup_14
+        >>> tsgroup_14.time_support
+
+        Index    rate
+        -------  ------
+              0     0.3
+             10     0.3
+
+                start    end
+        0       -5      5
+        shape: (1, 2), time unit: sec.
+
+        """
+        is_tsgroup = [isinstance(tsg, TsGroup) for tsg in tsgroups]
+        if not all(is_tsgroup):
+            not_tsgroup_index = [i+1 for i, boo in enumerate(is_tsgroup) if not boo]
+            raise TypeError(f"Passed variables at positions {not_tsgroup_index} are not TsGroup")
+
+        if len(tsgroups) == 1:
+            print('Only one TsGroup object provided, no merge needed')
+            return tsgroups[0]
+
+        tsg1    = tsgroups[0]
+        items    = tsg1.items()
+        keys     = set(tsg1.keys())
+        metadata = tsg1._metadata
+
+        for i, tsg in enumerate(tsgroups[1:]):
+            if not ignore_metadata:
+                if tsg1.metadata_columns != tsg.metadata_columns:
+                    raise ValueError(f"TsGroup at position {i+2} has different metadata columns from previous TsGroup objects. "
+                                    "Pass ignore_metadata=True to bypass")
+                metadata = pd.concat([metadata, tsg._metadata], axis=0)
+            
+            if not reset_index:
+                key_overlap = keys.intersection(tsg.keys())
+                if key_overlap:
+                    raise ValueError(f"TsGroup at position {i+2} has overlapping keys {key_overlap} with previous TsGroup objects. "
+                                     "Pass reset_index=True to bypass")
+                keys.update(tsg.keys())
+            
+            if reset_time_support:
+                time_support = None
+            else:
+                if not np.array_equal(
```

Comment:
You could be less "stringent" using `np.allclose` with `atol = 10**(-nap.config.time_index_precision)` and `rtol=0`. @gviejo, what do you think?

Reply:
I thought about `np.allclose` too. But to me the most ideal scenario of setting `reset_time_support=False` would be when the tsgroups were initiated with the same `time_support` and it's hard to imagine how small errors would arise from that. But open to your opinions too!

Reply:
Yes use allclose.
Thanks